src/dex/wusdm/wusdm.ts

```diff
@@ -0,0 +1,330 @@
+        isSell ? WUSDMFunctions.deposit : WUSDMFunctions.mint,
+        [isSell ? srcAmount : destAmount, this.augustusAddress],
+      );
+    } else {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.redeem : WUSDMFunctions.withdraw,
+        [
+          isSell ? srcAmount : destAmount,
+          this.augustusAddress,
+          this.augustusAddress,
+        ],
+      );
+    }
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      exchange,
+      undefined,
+      undefined,
+      undefined,
+      isSell && this.isUSDM(destToken),
+    );
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: WUSDMData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    const isSell = side === SwapSide.SELL;
+    const { exchange } = data;
+
+    let swapData: string;
+    if (this.isUSDM(srcToken)) {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.deposit : WUSDMFunctions.mint,
+        [isSell ? srcAmount : destAmount, recipient],
+      );
+    } else {
+      swapData = this.wUSDMInterface.encodeFunctionData(
+        isSell ? WUSDMFunctions.redeem : WUSDMFunctions.withdraw,
+        [isSell ? srcAmount : destAmount, recipient, executorAddress],
+      );
+    }
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: true,
+      exchangeData: swapData,
+      targetExchange: exchange,
+      returnAmountPos: isSell
+        ? extractReturnAmountPosition(
+            this.wUSDMInterface,
+            this.isUSDM(srcToken)
+              ? WUSDMFunctions.deposit
+              : WUSDMFunctions.redeem,
+          )
+        : undefined,
+      skipApproval: isSell && this.isUSDM(destToken),
+    };
+  }
+
+  // Encode params required by the exchange adapter
+  // V5: Used for multiSwap, buy & megaSwap
+  // V6: Not used, can be left blank
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: WUSDMData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { exchange } = data;
+
+    const payload = this.abiCoder.encodeParameter(
+      {
+        ParentStruct: {
+          toStaked: 'bool',
+        },
+      },
+      {
+        toStaked: this.isUSDM(srcToken),
+      },
+    );
+
+    return {
+      targetExchange: exchange,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // This is optional function in case if your implementation has acquired any resources
+  // you need to release for graceful shutdown. For example, it may be any interval timer
+  releaseResources(): AsyncOrSync<void> {
```

Comment:
@JackieJoo let's remove this empty method

Reply:
fixed